connect/runtime/src/main/java/org/apache/kafka/connect/runtime/errors/RetryWithToleranceOperator.java

```diff
@@ -331,6 +351,15 @@ public synchronized Throwable error() {
         return this.context.error();
     }
 
+    /**
+     * Exit from any currently ongoing retry loop and mark the operation as failed.
+     * This can be called from a separate thread to break out of an infinite retry loop in
+     * {@link #execAndRetry(Operation)}
+     */
+    public void exit() {
```

Comment:
Nit: I'm not a huge fan of `exit` since it has connotations of being able to immediately interrupt in-progress operations, which we don't provide at the moment. We could borrow from `WorkerTask` and call this `triggerStop` instead, and borrow from `AbstractWorkerSourceTask` and name the latch `stopRequestedLatch`.